Expand VEX data documentation (#48)
canonical/ubuntu-security-documentation: docs/security-updates/vex/index.rst

Ubuntu VEX data
###############

Ubuntu's Security Engineering Team publishes vulnerability data conforming to the structured, human and machine-readable Vulnerability Exploitability eXchange (VEX) format for all supported Ubuntu releases. In particular, the Ubuntu VEX
data feed follows the `OpenVEX <https://openvex.dev/>`_ specification, maintained by `OpenSSF <https://openssf.org/>`_. OpenVEX is an open source, minimal, compliant, interoperable, and embeddable implementation of VEX, that Canonical chose for clarity and ease of use.

As with the other vulnerability data feeds, Ubuntu's VEX data can be used in vulnerability and patch management processes.

The VEX data describes the exploitability status of known, publicly disclosed vulnerabilities, focusing on clarifying whether these are exploitable in
specific configurations.

The goal of the data is to provide users with additional information on whether a product is impacted by a specific vulnerability in an included component and, if affected, whether there are actions recommended to remediate the threat. 

> In many cases, a vulnerability in an upstream component will not be “exploitable” in the final product for various reasons, and this is the gap that VEX data will cover.

Reviewer: ```suggestion
In many cases, a vulnerability in an upstream component will not be “exploitable” in the final product for various reasons, and this is the gap that VEX data covers.
```

+---------------+--------------------------+------------------------------------------------------------------------------------+---------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| released      | Fixed                    | Canonical Security Engineering team has fixed the CVE and released the new version | fixed               | status_notes: A fix has been published. Additional notes provided if the fix is part of Ubuntu Pro (e.g., esm-infra, esm-apps).                                                  |
+---------------+--------------------------+------------------------------------------------------------------------------------+---------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

> The *CVE notes / justification* information are provided by the Ubuntu CVE Tracker and are being actively maintained by the Ubuntu Security Team.

Reviewer: ```suggestion
The *CVE notes / justification* information is provided by the Ubuntu CVE Tracker and is being actively maintained by the Ubuntu Security Team.
```

When it comes to USN (Ubuntu Security Notices) VEX data, the documents will only state the CVE entries that were fixed as a result of the update. Thus, it is expected that every statement in a USN VEX document will have a **fixed** status.

> The *status_notes* will indicate if the update is widely available, or if an Ubuntu Pro subscription is required.

Reviewer: ```suggestion
The *status_notes* will indicate if the update is widely available, or if you need an 
Ubuntu Pro subscription to get the update.
```
I think this phrasing (or something like this) is a bit more clear how the Ubuntu Pro subscription relates.

Limitations of the standard
===========================

One limitation of the OpenVEX standard is the lack of standard Severity field in the vulnerability metadata. This way, it is not possible for the user or system ingesting the data to know the severity and the prioritization of the remediation of the vulnerability.

> Canonical has overcome this limitation by adding more information in the *status_notes* field to inform about the severity of a CVE (following the `Ubuntu Priority schema <https://ubuntu.com/security/cves/about#priority>`_) in a programmatically parsable manner.

Reviewer 2: I think this sentence should reference the table above, to make it clear what it's referring to.

Author: This info is not included in the table, as prioritisation is a case-by-case basis mention in the VEX docs, and I wanted the table to be as generic as possible. Do you think we should tweak it somehow to include that information?

Using VEX data
==============

As a standardized, machine-friendly format, the VEX data feed can be used by any vulnerability management tool that supports it. More specifically, OpenSSF maintains `vexctl <https://github.com/openvex/vexctl>`_, an open source tool to create, apply and attest OpenVEX documents.

> Moreover, `grype <https://github.com/anchore/grype>`_, a vulnerability scanner for container images and filesystems by Anchore, can ingest OpenVEX data and provide more information regarding vulnerability status in a system.

Author: Why can't we? Its open source and public information

OpenVEX can also act as a complimentary to SBOMs (Software Bill of Materials) by providing precise metadata about vulnerability status of products mentioned in them. For example, `cycloneDX <https://cyclonedx.org/>`_ SBOMs support the integration of OpenVEX data.

Reporting issues in the data
============================

> If at any point you encounter inconsistencies with Ubuntu's VEX data, please report those by sending and email to security@ubuntu.com. We will gladly analyze and fix any issues.

Reviewer: I'd also suggest to bold the email address
